From df35c1841bf93d32995dc0df3752274cef80531c Mon Sep 17 00:00:00 2001
From: Paul Keith <javelinanddart@gmail.com>
Date: Sat, 2 Dec 2017 14:44:58 +0100
Subject: [PATCH 2/2] sdk: Remove low power restrictions on color control

* We no longer support the GPU backed, power hungry, color
  transformations that this check was put in place for, so
  remove the restriction on color temp changes for low power
* This is desirable because low power situations happen often
  at times later in the day, when blasting users with blue
  light (for auto night mode) is undesirable, and because
  reducing the intensity of colors is more power efficient,
  at least for AMOLED displays

Change-Id: I6b78a3f626cff3387d2acafa9c73cdf6af208fef
---
 .../internal/display/ColorTemperatureController.java   |  2 +-
 .../internal/display/DisplayHardwareController.java    | 10 ++++------
 2 files changed, 5 insertions(+), 7 deletions(-)

diff --git a/lineage/lib/main/java/org/lineageos/platform/internal/display/ColorTemperatureController.java b/lineage/lib/main/java/org/lineageos/platform/internal/display/ColorTemperatureController.java
index 17a4842..67853a4 100644
--- a/lineage/lib/main/java/org/lineageos/platform/internal/display/ColorTemperatureController.java
+++ b/lineage/lib/main/java/org/lineageos/platform/internal/display/ColorTemperatureController.java
@@ -187,7 +187,7 @@ public class ColorTemperatureController extends LiveDisplayFeature {
         int temperature = mDayTemperature;
         int mode = getMode();
 
-        if (mode == MODE_OFF || isLowPowerMode()) {
+        if (mode == MODE_OFF) {
             temperature = mDefaultDayTemperature;
         } else if (mode == MODE_NIGHT) {
             temperature = mNightTemperature;
diff --git a/lineage/lib/main/java/org/lineageos/platform/internal/display/DisplayHardwareController.java b/lineage/lib/main/java/org/lineageos/platform/internal/display/DisplayHardwareController.java
index ac5ce39..fe85dde 100644
--- a/lineage/lib/main/java/org/lineageos/platform/internal/display/DisplayHardwareController.java
+++ b/lineage/lib/main/java/org/lineageos/platform/internal/display/DisplayHardwareController.java
@@ -250,12 +250,10 @@ public class DisplayHardwareController extends LiveDisplayFeature {
 
         final float[] rgb = getDefaultAdjustment();
 
-        if (!isLowPowerMode()) {
-            copyColors(mColorAdjustment, rgb);
-            rgb[0] *= mAdditionalAdjustment[0];
-            rgb[1] *= mAdditionalAdjustment[1];
-            rgb[2] *= mAdditionalAdjustment[2];
-        }
+        copyColors(mColorAdjustment, rgb);
+        rgb[0] *= mAdditionalAdjustment[0];
+        rgb[1] *= mAdditionalAdjustment[1];
+        rgb[2] *= mAdditionalAdjustment[2];
 
         if (DEBUG) {
             Slog.d(TAG, "updateColorAdjustment: " + Arrays.toString(rgb));
-- 
2.17.0

